lib/aur-sync

```diff
@@ -101,124 +101,70 @@ usage() {
-            makechrootpkg_args+=(-D "$2")
-            shift 2 ;;
-        -T|--temp)
-            makechrootpkg_args+=(-T)
-            shift ;;
+        -c|--chroot)           chroot=1; build_args+=(-c) ;;
+        -f|--force)            build_args+=(-f) ;;
+        -s|--sign)             build_args+=(-sv) ;;
+        -D|--directory)        shift; build_args+=(-D "$1") ;;
+        --makepkg-conf)        shift; build_args+=(-M "$1") ;;
+        --pacman-conf)         shift; build_args+=(-C "$1") ;;
+        --bind)                shift; makechrootpkg_args+=(-D "$1") ;;
+        -T|--temp)             makechrootpkg_args+=(-T) ;;
         -A|--ignorearch|--ignore-arch)
-            makepkg_args+=(-A)
-            makechrootpkg_makepkg_args+=(-A)
-            shift ;;
-        -L|--log)
-            makepkg_args+=(-L)
-            shift ;;
+                               makepkg_args+=(-A);
+                               makechrootpkg_makepkg_args+=(-A) ;;
+        -L|--log)              makepkg_args+=(-L) ;;
         --noconfirm|--no-confirm)
-            makepkg_args+=(--noconfirm)
-            shift ;;
-        -l|--list)
-            list=1
-            shift ;;
-        --repo-list)
-            repo_list=1
-            shift ;;
-        -r|--rmdeps|--rm-deps)
-            makepkg_args+=(-r)
-            shift ;;
-        -p|--print)
-            build=0
-            shift ;;
-        -g|--git)
-            snapshot=0
-            shift ;;
-        -t|--tar)
-            snapshot=1
-            shift ;;
-        -u|--upgrades)
-            update=1
-            shift ;;
-        --allan|--nhehgvyf)
-            rotate=1
-            shift ;;
-        --continue)
-            download=0
-            shift ;;
-        --ignore)
-            IFS=, read -a pkg -r <<< "$2"
-            pkg_i+=("${pkg[@]}")
-            shift 2 ;;
-        --nover|--no-ver)
-            chkver=0
-            shift ;;
+                               makepkg_args+=(--noconfirm) ;;
+        -l|--list)             list=1 ;;
+        --repo-list)           repo_list=1 ;;
+        -r|--rmdeps|--rm-deps) makepkg_args+=(-r) ;;
+        -p|--print)            build=0 ;;
+        -g|--git)              snapshot=0 ;;
+        -t|--tar)              snapshot=1 ;;
+        -u|--upgrades)         update=1 ;;
+        --allan|--nhehgvyf)    rotate=1 ;;
+        --continue)            download=0 ;;
+        --ignore)              shift
+                               IFS=, read -a pkg -r <<< "$1"
+                               pkg_i+=("${pkg[@]}") ;;
+        --nover|--no-ver)      chkver=0 ;;
         --nover-shallow|--no-ver-shallow)
-            chkver_deps_only=1
-            shift ;;
-        --noview|--no-view)
-            view=0
-            shift ;;
-        --provides)
-            provides=1
-            shift ;;
-        -d|--database|--repo)
-            repo=$2
-            shift 2 ;;
-        --root)
-            root=$2
-            shift 2 ;;
-        *)
```

Comment:
Since I'm not yet familiar with `parseopts` - is `--` always guaranteed to be present in the option string, so that a glob is not required?

Reply:
I believe that's the case. c.f [parseopts.sh.in#n169][2] and [makepkg.sh.in#n1290][1]

[1]:https://git.archlinux.org/pacman.git/tree/scripts/libmakepkg/util/parseopts.sh.in?h=release/5.1.x#n169
[2]: https://git.archlinux.org/pacman.git/tree/scripts/makepkg.sh.in?h=release/5.1.x#n1289

Although, seems like makepkg.sh has a leftover `OPT_IND=0` from the previous option parser.

Reply:
Thanks. For OPT_IND, I'll blame @eli-schwartz. If the parser breaks, I'll blame him for it as well. :smile:

Reply:
Oh yes, blame poor Eli for everything. I see how this goes these days...

And after I gifted --packagelist to the world!

*goes off, sulks, breaks getopt to teach the world a lesson*